peps/pep-0734.rst

```diff
@@ -0,0 +1,399 @@
+   https://github.com/ericsnowcurrently/concurrency-benchmarks
+
+The module will include a basic mechanism for communicating between
+interpreters.  Without one, multiple interpreters are a much less
+useful feature.
+
+
+Specification
+=============
+
+The module will:
+
+* expose the existing multiple interpreter support
+* introduce a basic mechanism for communicating between interpreters
+
+The module will wrap a new low-level ``_interpreters`` module
+(in the same way as the ``threading`` module).  However, that low-level
+API is not intended for public use and thus not part of this proposal.
+
+We also expect that an ``InterpreterPoolExecutor`` will be added to the
+``concurrent.futures`` module, but that is outside the scope of this PEP.
+
+API: Using Interpreters
+-----------------------
+
+The module's top-level API for managing interpreters looks like this:
+
++----------------------------------+----------------------------------------------+
+| signature                        | description                                  |
++==================================+==============================================+
+| ``list_all() -> [Interpreter]``  | Get all existing interpreters.               |
++----------------------------------+----------------------------------------------+
+| ``get_current() -> Interpreter`` | Get the currently running interpreter.       |
++----------------------------------+----------------------------------------------+
+| ``create() -> Interpreter``      | Initialize a new (idle) Python interpreter.  |
++----------------------------------+----------------------------------------------+
+
+Each interpreter object:
+
++----------------------------------+------------------------------------------------+
+| signature                        | description                                    |
++==================================+================================================+
+| ``class Interpreter``            | A single interpreter.                          |
++----------------------------------+------------------------------------------------+
+| ``.id``                          | The interpreter's ID (read-only).              |
++----------------------------------+------------------------------------------------+
+| ``.is_running() -> bool``        | Is the interpreter currently executing code?   |
++----------------------------------+------------------------------------------------+
+| ``.set_main_attrs(**kwargs)``    | Bind objects in ``__main__``.                  |
++----------------------------------+------------------------------------------------+
+| ``.exec(code, /)``               | | Run the given source code in the interpreter |
+|                                  | | (in the current thread).                     |
++----------------------------------+------------------------------------------------+
+
+Additional details:
+
+* Every ``Interpreter`` instance wraps an ``InterpreterID`` object.
```

Comment:
Is this a public type? Is it the type of the `.id` attribute mentioned above?

Reply:
It is a public type and will be exposed as `interpreters.InterpreterID`.  However, I wasn't planning on including that in the PEP.  I'll reword the paragraph.